homeassistant/components/knx/__init__.py

```diff
@@ -251,16 +252,18 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             )
 
     hass.config_entries.async_setup_platforms(
-        entry, [platform for platform in SUPPORTED_PLATFORMS if platform in config]
+        entry,
+        [
+            platform
+            for platform in SUPPORTED_PLATFORMS
+            if platform in config and platform is not Platform.NOTIFY
+        ],
     )
 
-    # set up notify platform, no entry support for notify component yet,
-    # have to use discovery to load platform.
-    if NotifySchema.PLATFORM in conf:
+    # set up notify platform, no entry support for notify component yet
+    if NotifySchema.PLATFORM in config:
         hass.async_create_task(
-            discovery.async_load_platform(
-                hass, "notify", DOMAIN, conf[NotifySchema.PLATFORM], config
-            )
+            discovery.async_load_platform(hass, "notify", DOMAIN, {}, {})
```

Comment:
The last parameter should be the original config passed to `async_setup`.

Reply:
Whoops. Thanks @MartinHjelmare! Feel free to give me a tag if you intend to continue with the notify refactoring so I can give it a go if you need someone for testing.

Reply:
Yes, there's an open draft PR that I'll finish asap.